src/api/java/com/minecolonies/api/util/InventoryUtils.java

```diff
@@ -1776,21 +1776,16 @@ public static int transferXOfFirstSlotInItemHandlerWithIntoNextFreeSlotInItemHan
                 break;
             }
 
-
             final ItemStack returnStack = sourceHandler.extractItem(desiredItemSlot, currentAmount, false);
 
             if (!ItemStackUtils.isEmpty(returnStack))
             {
                 if (!InventoryUtils.addItemStackToItemHandler(targetHandler, returnStack))
                 {
-                    sourceHandler.insertItem(desiredItemSlot, returnStack, false);
                     break;
                 }
                 // Only reduce if successfully inserted.
-                else
-                {
-                    currentAmount -= returnStack.getCount();
-                }
+                currentAmount -= returnStack.getCount();
```

Comment:
think this should only reduce if the addItemStackToItemHandler is successfull

Reply:
If not, it breaks the loop.

Reply:
if !transfer
break

That's why.